### ISSUE TITLE
Carrot click triggers 'report opened' activity

### ISSUE THREAD

**Issue by blaughman** [usdoj-crt/crt-portal-management#1164]
## User story:
As a CRT user, I want a click on the view all table carrot to trigger the 'report opened' activity so that I can perform work in the view all table.

## Objective:
Increase efficiency for intake specialists

## Key result:
Time to open metric will be captured without going to the complaint detail page.

## Ticket background:
CRT users expressed frustration with having to open the complaint detail page to close reports from constant writers where it was blatantly obvious on the view all table that the new reports are the same subject matter as previous reports from that writer.  The GAO requirement is making staff less efficient when processing reports.

## Acceptance criteria:
1) Given I am on the view all table
when I click the carrot for a report to assess the 'personal description'
then the 'report opened' activity is captured for that report in the Activity Log

## Technical Notes/Questions:
- For bulk actions, we are limiting the 'bulk close' function to the 15 reports on the view all table #1157 
- We need UX assistance for the error state for bulk closing a report that hasn't been 'viewed/opened' #1157

**Comment by billyhunt** [comment]
## Checklist:

+ [ ] Go to the filter page "form/view", and click a down arrow.
+ [ ] Go to the admin page, and check the activity log.  Make sure that the activity log has a message like "Report viewed at 02/02/22 16:32:32 PM" with the correct timestamp.
+ [ ] In the admin page, check the report, and see if the "viewed" option is marked as true.
+ [ ] Go back to the filter page.  open that report, and make sure there is a log message saying something like "Report viewed at 02/02/22 16:32:32 PM" with the correct timestamp.  (When you clicked the arrow, not now)
+ [ ] Go to "/api" and see if you see something like this.  
![image](https://user-images.githubusercontent.com/6232068/152374369-367e331e-a5d4-4f82-866d-b124683f89e5.png)
+ [ ] From "/api", click the link (localhost or dev)"/api/reports/" and see if you see a page like this. 
![image](https://user-images.githubusercontent.com/6232068/152374603-35ed789b-5182-43f4-a042-d1bacd54781d.png)
+ [ ] From "/api/reports", click on a reports link (eg "/api/reports/1") and see if you see a page like this. 
![image](https://user-images.githubusercontent.com/6232068/152375029-46f4b6cd-54d7-47e9-b513-00432dc39482.png)
+ [ ] From "/api/reports/1", click the "viewed" checkbox in post and click "POST", the report should be marked as true.
+ [ ] Go to the admin page and make sure the log reflects that the report has been viewed.
+ [ ] Go to a different report (eg "/api/reports/2") and try the same thing with a "PUT"
+ [ ] Check the admin logs and see if report 2 viewed status has been logged.
+ [ ] Log out
+ [ ] Go to "/api/reports/1", you should see something like this.
![image](https://user-images.githubusercontent.com/6232068/152375837-cc68a04b-56bc-409d-98c2-a84cccdd7ab3.png)
+ [ ] Still logged out, go to "/api/reports", you should see something like this. 
![image](https://user-images.githubusercontent.com/6232068/152376016-daa7f233-e6be-4018-a6c9-409b240bb007.png)
+ [ ] Still logged out, go to "/api", you should see a login screen.
+ [ ] Login, you should be redirected to the "/api" page.

**Comment by blaughman** [comment]
@randyabramson @billyhunt @ilodidoj @louh I tested various scenarios and this appears to be working as intended.  I tested bulk actions and those reports where I clicked the carrot on the view all table are marking as 'report viewed.' I tested a group of three with two using the carrot and one was not clicked.  The two where it was clicked marked as viewed and the report not clicked did not show as viewed until I accessed the report detail page.  Long story short, this is working the way we expected it to function.
